### ISSUE TITLE
Stuck in a loop even after restore

### ISSUE THREAD

**Issue by YuckFouBTC** [raspiblitz/raspiblitz#4230]
**Describe the bug**
My node hardware (RasPi4) failed.
I saw something in the logs like a specific file failed checksum, so I mistakenly thought that it would help if I deleted the file.  Once I rm'ed the file, bitcoin's debug.log showed it kept looping, complaining about that specific file missing.
Replaced node with a new RasPi4.
Flashed new SDHC with latest raspiblitz.
Booted up and it saw the files on the SSD, asked me if I wanted to restore, I agreed.  Great, I thought.  Nope...
It's now looping again on the same place it was before I swapped out the hardware and flashed a new SDHC.

**To Reproduce**
On a node you don't care about, do this;
`sudo ls -la /mnt/hdd/bitcoin/indexes/txindex/`
Choose one of the files, then delete it like this;
`sudo rm /mnt/hdd/bitcoin/indexes/txindex/053716.ldb`
then reboot the node and see the looping failure on that missing file with no obvious solution.

**Debug file**
[loopy.txt](https://github.com/raspiblitz/raspiblitz/files/13183049/loopy.txt) | https://termbin.com/101t | https://archive.ph/rGxEc

Please add something to the code to help idiots like me figure out how to get out of this stuck-in-loop situation, or add comment(s) on suggestions for what I can try here.  I love the raspiblitz software and commend you on producing the best bitcoin lightning node software I have ever seen.

**Comment by YuckFouBTC** [comment]
Decided to try harder at fixing this myself.

I have another raspiblitz node on my LAN that is working just fine.
I will use that node's non-corrupted blockchain data to copy over to my stuck-in-loop node.

192.168.1.247 is the IP address of my stuck-in-loop node;
```
admin@192.168.1.247:~ ₿ sudo systemctl stop lnd
admin@192.168.1.247:~ ₿ sudo systemctl stop bitcoind
admin@192.168.1.247:~ ₿ sudo rm -rf /mnt/hdd/bitcoin/blocks
admin@192.168.1.247:~ ₿ sudo rm -rf /mnt/hdd/bitcoin/chainstate
admin@192.168.1.247:~ ₿ 

```

I then went to my good raspiblitz node: MENU > REPAIR > COPY-SOURCE.

It's now in the process of copying the data, 1% completed.  ETA = ~4 hours.

**Comment by YuckFouBTC** [comment]
26% done...

**Comment by YuckFouBTC** [comment]
Looks like it's done.

Now seeing this:
![Screenshot from 2023-10-27 07-50-41](https://github.com/raspiblitz/raspiblitz/assets/115867254/72063320-015a-49f1-8db3-e014027b2924)

**Comment by YuckFouBTC** [comment]
Did a ^C to get out of that screen, followed by:
`sudo reboot`

Gave it a few minutes, ssh'ed back in.  Now seeing the same problem still;
![Screenshot from 2023-10-27 07-57-33](https://github.com/raspiblitz/raspiblitz/assets/115867254/185a1ae0-b327-4433-a4ce-52a3efd8733b)

**Comment by YuckFouBTC** [comment]
The same stuck-in-loop problem still exists;

admin@192.168.1.247:~ ₿ sudo tail -f /mnt/hdd/bitcoin/debug.log
2023-10-27T07:58:02Z * Using 2230.2 MiB for in-memory UTXO set (plus up to 286.1 MiB of unused mempool space)
2023-10-27T07:58:02Z init message: Loading block index…
2023-10-27T07:58:02Z Assuming ancestors of block 000000000000000000035c3f0d31e71a5ee24c5aaf3354689f65bd7b07dee632 have valid signatures.
2023-10-27T07:58:02Z Setting nMinimumChainWork=000000000000000000000000000000000000000044a50fe819c39ad624021859
2023-10-27T07:58:02Z Opening LevelDB in /mnt/hdd/bitcoin/blocks/index
2023-10-27T07:58:02Z Opened LevelDB successfully
2023-10-27T07:58:02Z Using obfuscation key for /mnt/hdd/bitcoin/blocks/index: 0000000000000000
2023-10-27T07:58:14Z LoadBlockIndexDB: last block file = 3904
2023-10-27T07:58:14Z LoadBlockIndexDB: last block file info: CBlockFileInfo(blocks=3, size=4641724, heights=813976...813978, time=2023-10-26...2023-10-26)
2023-10-27T07:58:14Z Checking all blk files are present...
2023-10-27T07:58:18Z Initializing chainstate Chainstate [ibd] @ height -1 (null)
2023-10-27T07:58:18Z Opening LevelDB in /mnt/hdd/bitcoin/chainstate
2023-10-27T07:58:18Z Opened LevelDB successfully
2023-10-27T07:58:18Z Using obfuscation key for /mnt/hdd/bitcoin/chainstate: 7424f602662de5fb
2023-10-27T07:58:19Z Loaded best chain: hashBestChain=0000000000000000000104dae34a02d245d6e18316e1ad8035e504fc8dc006d6 height=813978 date=2023-10-26T22:03:08Z progress=0.999856
2023-10-27T07:58:19Z [snapshot] allocating all cache to the IBD chainstate
2023-10-27T07:58:19Z Opening LevelDB in /mnt/hdd/bitcoin/chainstate
2023-10-27T07:58:19Z Opened LevelDB successfully
2023-10-27T07:58:19Z Using obfuscation key for /mnt/hdd/bitcoin/chainstate: 7424f602662de5fb
2023-10-27T07:58:19Z [Chainstate [ibd] @ height 813978 (0000000000000000000104dae34a02d245d6e18316e1ad8035e504fc8dc006d6)] resized coinsdb cache to 8.0 MiB
2023-10-27T07:58:19Z [Chainstate [ibd] @ height 813978 (0000000000000000000104dae34a02d245d6e18316e1ad8035e504fc8dc006d6)] resized coinstip cache to 2230.2 MiB
2023-10-27T07:58:19Z init message: Verifying blocks…
2023-10-27T07:58:19Z Verifying last 6 blocks at level 3
2023-10-27T07:58:19Z Verification progress: 0%
2023-10-27T07:58:20Z Verification progress: 16%
2023-10-27T07:58:21Z Verification progress: 33%
2023-10-27T07:58:22Z Verification progress: 50%
2023-10-27T07:58:23Z Verification progress: 66%
2023-10-27T07:58:24Z Verification progress: 83%
2023-10-27T07:58:25Z Verification progress: 99%
2023-10-27T07:58:25Z Verification: No coin database inconsistencies in last 6 blocks (16152 transactions)
2023-10-27T07:58:25Z  block index           23787ms
2023-10-27T07:58:25Z Opening LevelDB in /mnt/hdd/bitcoin/indexes/txindex
2023-10-27T07:58:25Z Fatal LevelDB error: Corruption: 1 missing files; e.g.: /mnt/hdd/bitcoin/indexes/txindex/053716.ldb
2023-10-27T07:58:25Z You can use -debug=leveldb to get more complete diagnostic messages
2023-10-27T07:58:25Z 

************************
EXCEPTION: 15dbwrapper_error       
Fatal LevelDB error: Corruption: 1 missing files; e.g.: /mnt/hdd/bitcoin/indexes/txindex/053716.ldb       
bitcoin in AppInit()       

2023-10-27T07:58:25Z Shutdown: In progress...
2023-10-27T07:58:25Z scheduler thread exit
2023-10-27T07:58:26Z Shutdown: done





2023-10-27T07:58:27Z Bitcoin Core version v25.0.0 (release build)
2023-10-27T07:58:27Z InitParameterInteraction: parameter interaction: -bind set -> setting -listen=1
2023-10-27T07:58:27Z InitParameterInteraction: parameter interaction: -proxy set -> setting -upnp=0
2023-10-27T07:58:27Z InitParameterInteraction: parameter interaction: -proxy set -> setting -natpmp=0
2023-10-27T07:58:27Z InitParameterInteraction: parameter interaction: -proxy set -> setting -discover=0
2023-10-27T07:58:27Z Using the 'standard' SHA256 implementation
2023-10-27T07:58:27Z Default data directory /home/bitcoin/.bitcoin
2023-10-27T07:58:27Z Using data directory /mnt/hdd/bitcoin
2023-10-27T07:58:27Z Config file: /mnt/hdd/bitcoin/bitcoin.conf
2023-10-27T07:58:27Z Config file arg: daemon="1"
2023-10-27T07:58:27Z Config file arg: datadir="/mnt/hdd/bitcoin"
2023-10-27T07:58:27Z Config file arg: dbcache="2560"
2023-10-27T07:58:27Z Config file arg: deprecatedrpc="addresses"
2023-10-27T07:58:27Z Config file arg: disablewallet="1"
2023-10-27T07:58:27Z Config file arg: dns="0"
2023-10-27T07:58:27Z Config file arg: dnsseed="0"
2023-10-27T07:58:27Z Config file arg: maxconnections="40"
2023-10-27T07:58:27Z Config file arg: maxmempool="300"
2023-10-27T07:58:27Z Config file arg: maxorphantx="10"
2023-10-27T07:58:27Z Config file arg: maxuploadtarget="5000"
2023-10-27T07:58:27Z Config file arg: onlynet="onion"
2023-10-27T07:58:27Z Config file arg: peerbloomfilters="1"
2023-10-27T07:58:27Z Config file arg: proxy="127.0.0.1:9050"
2023-10-27T07:58:27Z Config file arg: rpcallowip="127.0.0.1"
2023-10-27T07:58:27Z Config file arg: rpcpassword=****
2023-10-27T07:58:27Z Config file arg: rpcuser=****
2023-10-27T07:58:27Z Config file arg: server="1"
2023-10-27T07:58:27Z Config file arg: testnet="0"
2023-10-27T07:58:27Z Config file arg: txindex="1"
2023-10-27T07:58:27Z Config file arg: [main] bind="127.0.0.1"
2023-10-27T07:58:27Z Config file arg: [main] debuglogfile="/mnt/hdd/bitcoin/debug.log"
2023-10-27T07:58:27Z Config file arg: [main] rpcbind="127.0.0.1:8332"
2023-10-27T07:58:27Z Config file arg: [main] rpcport="8332"
2023-10-27T07:58:27Z Config file arg: [main] zmqpubrawblock="tcp://127.0.0.1:28332"
2023-10-27T07:58:27Z Config file arg: [main] zmqpubrawtx="tcp://127.0.0.1:28333"
2023-10-27T07:58:27Z Config file arg: [signet] debuglogfile="/mnt/hdd/bitcoin/signet/debug.log"
2023-10-27T07:58:27Z Config file arg: [test] bind="127.0.0.1"
2023-10-27T07:58:27Z Config file arg: [test] debuglogfile="/mnt/hdd/bitcoin/testnet3/debug.log"
2023-10-27T07:58:27Z Config file arg: [test] rpcbind="127.0.0.1:18332"
2023-10-27T07:58:27Z Config file arg: [test] rpcport="18332"
2023-10-27T07:58:27Z Command-line arg: conf="/mnt/hdd/bitcoin/bitcoin.conf"
2023-10-27T07:58:27Z Command-line arg: daemonwait=""
2023-10-27T07:58:27Z Command-line arg: datadir="/mnt/hdd/bitcoin"
2023-10-27T07:58:27Z Using at most 40 automatic connections (1024 file descriptors available)
2023-10-27T07:58:27Z Using 16 MiB out of 16 MiB requested for signature cache, able to store 524288 elements
2023-10-27T07:58:27Z Using 16 MiB out of 16 MiB requested for script execution cache, able to store 524288 elements
2023-10-27T07:58:27Z Script verification uses 3 additional threads
2023-10-27T07:58:27Z Wallet disabled!
2023-10-27T07:58:27Z scheduler thread start
2023-10-27T07:58:27Z Binding RPC on address 127.0.0.1 port 8332
2023-10-27T07:58:27Z [http] creating work queue of depth 16
2023-10-27T07:58:27Z Config options rpcuser and rpcpassword will soon be deprecated. Locally-run instances may remove rpcuser to use cookie-based auth, or may be replaced with rpcauth. Please see share/rpcauth for rpcauth auth generation.
2023-10-27T07:58:27Z [http] starting 4 worker threads
2023-10-27T07:58:27Z Using /16 prefix for IP bucketing
2023-10-27T07:58:27Z init message: Loading P2P addresses…
2023-10-27T07:58:27Z Loaded 10541 addresses from peers.dat  163ms
2023-10-27T07:58:27Z init message: Loading banlist…
2023-10-27T07:58:27Z SetNetworkActive: true
2023-10-27T07:58:27Z Cache configuration:
2023-10-27T07:58:27Z * Using 2.0 MiB for block index database
2023-10-27T07:58:27Z * Using 319.8 MiB for transaction index database
2023-10-27T07:58:27Z * Using 8.0 MiB for chain state database
2023-10-27T07:58:27Z * Using 2230.2 MiB for in-memory UTXO set (plus up to 286.1 MiB of unused mempool space)
2023-10-27T07:58:27Z init message: Loading block index…
2023-10-27T07:58:27Z Assuming ancestors of block 000000000000000000035c3f0d31e71a5ee24c5aaf3354689f65bd7b07dee632 have valid signatures.
2023-10-27T07:58:27Z Setting nMinimumChainWork=000000000000000000000000000000000000000044a50fe819c39ad624021859
2023-10-27T07:58:27Z Opening LevelDB in /mnt/hdd/bitcoin/blocks/index
2023-10-27T07:58:27Z Opened LevelDB successfully
2023-10-27T07:58:27Z Using obfuscation key for /mnt/hdd/bitcoin/blocks/index: 0000000000000000
2023-10-27T07:58:40Z LoadBlockIndexDB: last block file = 3904
2023-10-27T07:58:40Z LoadBlockIndexDB: last block file info: CBlockFileInfo(blocks=3, size=4641724, heights=813976...813978, time=2023-10-26...2023-10-26)
2023-10-27T07:58:40Z Checking all blk files are present...
2023-10-27T07:58:43Z Initializing chainstate Chainstate [ibd] @ height -1 (null)
2023-10-27T07:58:43Z Opening LevelDB in /mnt/hdd/bitcoin/chainstate
2023-10-27T07:58:44Z Opened LevelDB successfully
2023-10-27T07:58:44Z Using obfuscation key for /mnt/hdd/bitcoin/chainstate: 7424f602662de5fb
2023-10-27T07:58:44Z Loaded best chain: hashBestChain=0000000000000000000104dae34a02d245d6e18316e1ad8035e504fc8dc006d6 height=813978 date=2023-10-26T22:03:08Z progress=0.999856
2023-10-27T07:58:44Z [snapshot] allocating all cache to the IBD chainstate
2023-10-27T07:58:44Z Opening LevelDB in /mnt/hdd/bitcoin/chainstate
2023-10-27T07:58:44Z Opened LevelDB successfully
2023-10-27T07:58:44Z Using obfuscation key for /mnt/hdd/bitcoin/chainstate: 7424f602662de5fb
2023-10-27T07:58:44Z [Chainstate [ibd] @ height 813978 (0000000000000000000104dae34a02d245d6e18316e1ad8035e504fc8dc006d6)] resized coinsdb cache to 8.0 MiB
2023-10-27T07:58:44Z [Chainstate [ibd] @ height 813978 (0000000000000000000104dae34a02d245d6e18316e1ad8035e504fc8dc006d6)] resized coinstip cache to 2230.2 MiB
2023-10-27T07:58:44Z init message: Verifying blocks…
2023-10-27T07:58:44Z Verifying last 6 blocks at level 3
2023-10-27T07:58:44Z Verification progress: 0%
2023-10-27T07:58:45Z Verification progress: 16%
2023-10-27T07:58:46Z Verification progress: 33%
2023-10-27T07:58:48Z Verification progress: 50%
2023-10-27T07:58:49Z Verification progress: 66%
2023-10-27T07:58:50Z Verification progress: 83%
2023-10-27T07:58:51Z Verification progress: 99%
2023-10-27T07:58:51Z Verification: No coin database inconsistencies in last 6 blocks (16152 transactions)
2023-10-27T07:58:51Z  block index           23765ms
2023-10-27T07:58:51Z Opening LevelDB in /mnt/hdd/bitcoin/indexes/txindex
2023-10-27T07:58:51Z Fatal LevelDB error: Corruption: 1 missing files; e.g.: /mnt/hdd/bitcoin/indexes/txindex/053716.ldb
2023-10-27T07:58:51Z You can use -debug=leveldb to get more complete diagnostic messages
2023-10-27T07:58:51Z 

************************
EXCEPTION: 15dbwrapper_error       
Fatal LevelDB error: Corruption: 1 missing files; e.g.: /mnt/hdd/bitcoin/indexes/txindex/053716.ldb       
bitcoin in AppInit()       

2023-10-27T07:58:51Z Shutdown: In progress...
2023-10-27T07:58:51Z scheduler thread exit
2023-10-27T07:58:52Z Shutdown: done





2023-10-27T07:58:52Z Bitcoin Core version v25.0.0 (release build)
2023-10-27T07:58:52Z InitParameterInteraction: parameter interaction: -bind set -> setting -listen=1
2023-10-27T07:58:52Z InitParameterInteraction: parameter interaction: -proxy set -> setting -upnp=0
2023-10-27T07:58:52Z InitParameterInteraction: parameter interaction: -proxy set -> setting -natpmp=0
2023-10-27T07:58:52Z InitParameterInteraction: parameter interaction: -proxy set -> setting -discover=0
2023-10-27T07:58:52Z Using the 'standard' SHA256 implementation
2023-10-27T07:58:52Z Default data directory /home/bitcoin/.bitcoin
2023-10-27T07:58:52Z Using data directory /mnt/hdd/bitcoin
2023-10-27T07:58:52Z Config file: /mnt/hdd/bitcoin/bitcoin.conf
2023-10-27T07:58:52Z Config file arg: daemon="1"
2023-10-27T07:58:52Z Config file arg: datadir="/mnt/hdd/bitcoin"
2023-10-27T07:58:52Z Config file arg: dbcache="2560"
2023-10-27T07:58:52Z Config file arg: deprecatedrpc="addresses"
2023-10-27T07:58:52Z Config file arg: disablewallet="1"
2023-10-27T07:58:52Z Config file arg: dns="0"
2023-10-27T07:58:52Z Config file arg: dnsseed="0"
2023-10-27T07:58:52Z Config file arg: maxconnections="40"
2023-10-27T07:58:52Z Config file arg: maxmempool="300"
2023-10-27T07:58:52Z Config file arg: maxorphantx="10"
2023-10-27T07:58:52Z Config file arg: maxuploadtarget="5000"
2023-10-27T07:58:52Z Config file arg: onlynet="onion"
2023-10-27T07:58:52Z Config file arg: peerbloomfilters="1"
2023-10-27T07:58:52Z Config file arg: proxy="127.0.0.1:9050"
2023-10-27T07:58:52Z Config file arg: rpcallowip="127.0.0.1"
2023-10-27T07:58:52Z Config file arg: rpcpassword=****
2023-10-27T07:58:52Z Config file arg: rpcuser=****
2023-10-27T07:58:52Z Config file arg: server="1"
2023-10-27T07:58:52Z Config file arg: testnet="0"
2023-10-27T07:58:52Z Config file arg: txindex="1"
2023-10-27T07:58:52Z Config file arg: [main] bind="127.0.0.1"
2023-10-27T07:58:52Z Config file arg: [main] debuglogfile="/mnt/hdd/bitcoin/debug.log"
2023-10-27T07:58:52Z Config file arg: [main] rpcbind="127.0.0.1:8332"
2023-10-27T07:58:52Z Config file arg: [main] rpcport="8332"
2023-10-27T07:58:52Z Config file arg: [main] zmqpubrawblock="tcp://127.0.0.1:28332"
2023-10-27T07:58:52Z Config file arg: [main] zmqpubrawtx="tcp://127.0.0.1:28333"
2023-10-27T07:58:52Z Config file arg: [signet] debuglogfile="/mnt/hdd/bitcoin/signet/debug.log"
2023-10-27T07:58:52Z Config file arg: [test] bind="127.0.0.1"
2023-10-27T07:58:52Z Config file arg: [test] debuglogfile="/mnt/hdd/bitcoin/testnet3/debug.log"
2023-10-27T07:58:52Z Config file arg: [test] rpcbind="127.0.0.1:18332"
2023-10-27T07:58:52Z Config file arg: [test] rpcport="18332"
2023-10-27T07:58:52Z Command-line arg: conf="/mnt/hdd/bitcoin/bitcoin.conf"
2023-10-27T07:58:52Z Command-line arg: daemonwait=""
2023-10-27T07:58:52Z Command-line arg: datadir="/mnt/hdd/bitcoin"
2023-10-27T07:58:52Z Using at most 40 automatic connections (1024 file descriptors available)
2023-10-27T07:58:52Z Using 16 MiB out of 16 MiB requested for signature cache, able to store 524288 elements
2023-10-27T07:58:52Z Using 16 MiB out of 16 MiB requested for script execution cache, able to store 524288 elements
2023-10-27T07:58:52Z Script verification uses 3 additional threads
2023-10-27T07:58:52Z Wallet disabled!
2023-10-27T07:58:52Z scheduler thread start
2023-10-27T07:58:52Z Binding RPC on address 127.0.0.1 port 8332
2023-10-27T07:58:52Z [http] creating work queue of depth 16
2023-10-27T07:58:52Z Config options rpcuser and rpcpassword will soon be deprecated. Locally-run instances may remove rpcuser to use cookie-based auth, or may be replaced with rpcauth. Please see share/rpcauth for rpcauth auth generation.
2023-10-27T07:58:52Z [http] starting 4 worker threads
2023-10-27T07:58:52Z Using /16 prefix for IP bucketing
2023-10-27T07:58:52Z init message: Loading P2P addresses…
2023-10-27T07:58:52Z Loaded 10541 addresses from peers.dat  177ms
2023-10-27T07:58:52Z init message: Loading banlist…
2023-10-27T07:58:52Z SetNetworkActive: true
2023-10-27T07:58:52Z Cache configuration:
2023-10-27T07:58:52Z * Using 2.0 MiB for block index database
2023-10-27T07:58:52Z * Using 319.8 MiB for transaction index database
2023-10-27T07:58:52Z * Using 8.0 MiB for chain state database
2023-10-27T07:58:52Z * Using 2230.2 MiB for in-memory UTXO set (plus up to 286.1 MiB of unused mempool space)
2023-10-27T07:58:52Z init message: Loading block index…
2023-10-27T07:58:52Z Assuming ancestors of block 000000000000000000035c3f0d31e71a5ee24c5aaf3354689f65bd7b07dee632 have valid signatures.
2023-10-27T07:58:52Z Setting nMinimumChainWork=000000000000000000000000000000000000000044a50fe819c39ad624021859
2023-10-27T07:58:52Z Opening LevelDB in /mnt/hdd/bitcoin/blocks/index
2023-10-27T07:58:52Z Opened LevelDB successfully
2023-10-27T07:58:52Z Using obfuscation key for /mnt/hdd/bitcoin/blocks/index: 0000000000000000
^C
admin@192.168.1.247:~ ₿

**Comment by YuckFouBTC** [comment]
Trying this;

```
admin@192.168.1.247:~ ₿ sudo -u bitcoin /usr/local/bin/bitcoind -reindex
Bitcoin Core starting
admin@192.168.1.247:~ ₿ 

```

**Comment by YuckFouBTC** [comment]
There are 1905 files that need to be reindexed.

```
admin@192.168.1.247:~ ₿ sudo ls -la /mnt/hdd/bitcoin/blocks/ | grep blk0 | wc -l
3905
admin@192.168.1.247:~ ₿ 
```

Watching the action here;
`admin@192.168.1.247:~ ₿ sudo tail -n 1000 -f /mnt/hdd/bitcoin/debug.log`

Looks like this is going to take a few hours...

**Comment by YuckFouBTC** [comment]
There are 3905 files that need to be reindexed.

admin@192.168.1.247:~ ₿ sudo ls -la /mnt/hdd/bitcoin/blocks/ | grep blk0 | wc -l
3905
admin@192.168.1.247:~ ₿ 

Watching the action here;
`admin@192.168.1.247:~ ₿ sudo tail -n 1000 -f /mnt/hdd/bitcoin/debug.log`

Looks like this is going to take a long time...

**Comment by YuckFouBTC** [comment]
There are 3905 files that need to be reindexed.

```
admin@192.168.1.247:~ ₿ sudo ls -la /mnt/hdd/bitcoin/blocks/ | grep blk0 | wc -l
3905
admin@192.168.1.247:~ ₿

```
Watching the action here;
`admin@192.168.1.247:~ ₿ sudo tail -n 1000 -f /mnt/hdd/bitcoin/debug.log`

Looks like this is going to take a long time...

**Comment by YuckFouBTC** [comment]
Progress update: still going...

2023-10-27T14:28:28Z Loaded 47 blocks from external file in 13166ms
2023-10-27T14:28:28Z Reindexing block file blk01397.dat...
2023-10-27T14:28:50Z Loaded 175 blocks from external file in 21253ms
2023-10-27T14:28:50Z Reindexing block file blk01398.dat...
2023-10-27T14:29:22Z Loaded 226 blocks from external file in 32099ms
2023-10-27T14:29:22Z Reindexing block file blk01399.dat...
2023-10-27T14:29:40Z Loaded 120 blocks from external file in 18813ms
2023-10-27T14:29:40Z Reindexing block file blk01400.dat...
2023-10-27T14:30:01Z Loaded 151 blocks from external file in 20839ms
2023-10-27T14:30:01Z Reindexing block file blk01401.dat...
2023-10-27T14:30:21Z Loaded 129 blocks from external file in 20138ms
2023-10-27T14:30:21Z Reindexing block file blk01402.dat...
2023-10-27T14:30:45Z Loaded 145 blocks from external file in 23109ms
2023-10-27T14:30:45Z Reindexing block file blk01403.dat...
2023-10-27T14:31:04Z Loaded 127 blocks from external file in 19657ms
2023-10-27T14:31:04Z Reindexing block file blk01404.dat...

**Comment by YuckFouBTC** [comment]
Progress update: 3266 blk0*.dat files reindexed out of 3905 total.  So 3266 * 3950 = 0.836363636 * 100 = 83.6363636% completed

**Comment by YuckFouBTC** [comment]
Progress update: 3266 blk0*.dat files reindexed out of 3905 total.  So 3266 / 3950 = 0.836363636 * 100 = 83.6363636% completed

**Comment by YuckFouBTC** [comment]
Progress update: it looks like all 3,266 of the blk0*.dat files were processed.

The reindex has moved on to the next phase.  This part is going super slow.  Looks like it will be a couple days yet...

This is what I see when doing `sudo tail -n 1000 -f /mnt/hdd/bitcoin/debug.log`:
2023-10-28T21:00:26Z UpdateTip: new best=000000000000000000b6f29a8a88b5418c3cc5abf17a15d43d8e1a06a1cef703 height=426013 version=0x20000000 log2_work=85.155665 tx=150547940 date='2016-08-20T08:49:49Z' progress=0.169829 cache=895.3MiB(5794686txo)
2023-10-28T21:00:27Z UpdateTip: new best=000000000000000004508f3efeb9b814dd5c9a6984bf695ed22ab86af7b24d26 height=426014 version=0x20000000 log2_work=85.155696 tx=150550075 date='2016-08-20T09:04:11Z' progress=0.169831 cache=895.4MiB(5795794txo)
2023-10-28T21:00:28Z UpdateTip: new best=000000000000000003124a980ca59758bb4d952229986a34057536651291aac3 height=426015 version=0x20000000 log2_work=85.155727 tx=150551887 date='2016-08-20T09:16:38Z' progress=0.169833 cache=895.5MiB(5795960txo)
2023-10-28T21:00:28Z BlockUntilSyncedToCurrentChain: txindex is catching up on block notifications
2023-10-28T21:00:28Z UpdateTip: new best=0000000000000000039f100b08d8821dae5767b198966f7cbfe0c0ddc7be05d5 height=426016 version=0x20000000 log2_work=85.155758 tx=150552648 date='2016-08-20T09:20:38Z' progress=0.169834 cache=895.5MiB(5796173txo)
2023-10-28T21:00:33Z UpdateTip: new best=00000000000000000249d9ff629bf9a864871d081130bf4a12cf9eef8473129d height=426017 version=0x20000000 log2_work=85.155790 tx=150553342 date='2016-08-20T09:23:35Z' progress=0.169835 cache=895.8MiB(5799033txo)
2023-10-28T21:00:34Z UpdateTip: new best=00000000000000000250e79f84879993549008345e2e24f72c4212c7cd0a6202 height=426018 version=0x20000000 log2_work=85.155821 tx=150555894 date='2016-08-20T09:44:35Z' progress=0.169838 cache=896.0MiB(5800113txo)
2023-10-28T21:00:38Z UpdateTip: new best=00000000000000000270bb32f7d42ae03742bbc9124d349e10774d8cd7a57856 height=426019 version=0x20000000 log2_work=85.155852 tx=150556621 date='2016-08-20T09:45:39Z' progress=0.169839 cache=896.2MiB(5801650txo)
2023-10-28T21:00:38Z UpdateTip: new best=000000000000000000733817ff6a52cdea1ab4273bc68cfd55c223405818551c height=426020 version=0x20000000 log2_work=85.155883 tx=150557802 date='2016-08-20T09:54:20Z' progress=0.169840 cache=896.3MiB(5802549txo)
2023-10-28T21:00:39Z UpdateTip: new best=0000000000000000010a6212570a433c98b5b1f298a953f785a45ae9d1cb4c73 height=426021 version=0x20000000 log2_work=85.155915 tx=150559010 date='2016-08-20T10:02:35Z' progress=0.169841 cache=896.5MiB(5804306txo)
2023-10-28T21:00:39Z BlockUntilSyncedToCurrentChain: txindex is catching up on block notifications
2023-10-28T21:00:41Z UpdateTip: new best=0000000000000000014392b28167efe78bb182d28b19a06b36f623833e1fd077 height=426022 version=0x20000000 log2_work=85.155946 tx=150560187 date='2016-08-20T10:09:26Z' progress=0.169843 cache=897.2MiB(5809760txo)
2023-10-28T21:00:42Z UpdateTip: new best=000000000000000001e7f96119eb022bfb5c82490a463022678bfd91a9471cb1 height=426023 version=0x20000000 log2_work=85.155977 tx=150563077 date='2016-08-20T10:38:34Z' progress=0.169846 cache=897.4MiB(5811763txo)
2023-10-28T21:00:43Z UpdateTip: new best=000000000000000003cf30612f29d23e7e5cf98390ec4ef5faab18a2cd68a57b height=426024 version=0x20000000 log2_work=85.156008 tx=150565544 date='2016-08-20T10:52:03Z' progress=0.169849 cache=897.6MiB(5813747txo)
2023-10-28T21:00:44Z UpdateTip: new best=0000000000000000007da31c882a318810f4f66812ed692d4e166fd7cd25f5d3 height=426025 version=0x20000000 log2_work=85.156040 tx=150567624 date='2016-08-20T11:09:12Z' progress=0.169851 cache=897.6MiB(5813781txo)
2023-10-28T21:00:45Z UpdateTip: new best=000000000000000003037d050fa86cdd17b8e0c670550e4f81c9b5cf5f5debb4 height=426026 version=0x20000000 log2_work=85.156071 tx=150569961 date='2016-08-20T11:37:47Z' progress=0.169854 cache=897.8MiB(5814663txo)
2023-10-28T21:00:46Z UpdateTip: new best=0000000000000000042feb02dc744796cdd3d04f011f1ebf841707b7f5d5b86a height=426027 version=0x20000000 log2_work=85.156102 tx=150572707 date='2016-08-20T11:50:34Z' progress=0.169857 cache=897.9MiB(5816120txo)
2023-10-28T21:00:46Z BlockUntilSyncedToCurrentChain: txindex is catching up on block notifications
2023-10-28T21:00:46Z UpdateTip: new best=000000000000000000e90fa028a2fe7bffdc1fd43039ea32be9c476488e34841 height=426028 version=0x20000000 log2_work=85.156133 tx=150574048 date='2016-08-20T11:55:48Z' progress=0.169858 cache=897.7MiB(5813995txo)
2023-10-28T21:00:47Z UpdateTip: new best=000000000000000001d4a8dc1ec8b2562a0242d03b878ee293e8b3c9391b8abf height=426029 version=0x20000000 log2_work=85.156165 tx=150575905 date='2016-08-20T12:04:14Z' progress=0.169860 cache=897.7MiB(5814118txo)
2023-10-28T21:00:48Z UpdateTip: new best=0000000000000000040cc385a1ba9f582d66186a5415a1e666b2478e1b281782 height=426030 version=0x20000000 log2_work=85.156196 tx=150577504 date='2016-08-20T12:13:48Z' progress=0.169862 cache=898.7MiB(5822579txo)
2023-10-28T21:00:48Z BlockUntilSyncedToCurrentChain: txindex is catching up on block notifications

**Comment by YuckFouBTC** [comment]
I am going to Close this issue since it's quite obvious to me now that the corruption issue that I encountered was not a problem with Raspiblitz.

The only thing that I can think of that could possibly done would be for raspiblitz to give an informative error message about what the problem is if it sees a file missing like what happened with me after I deleted that file.

The error message that raspiblitz could look for would match this regex when parsing bitcoin's debug.log:
/^Fatal LevelDB error: Corruption: (\d+) missing files; e.g.: \/mnt\/hdd\/bitcoin\/indexes\/txindex\/(\d+).ldb$/

If such a match was found, the recommended action would be to say that a critical file was missing and that a reindex may be necessary to correct it.

**Comment by YuckFouBTC** [comment]
I am going to Close this issue since it's quite obvious to me now that the corruption issue that I encountered was not a problem with Raspiblitz.

The only thing that I can think of that could possibly be done would be for raspiblitz to give an informative error message about what the problem is if it sees a file missing like what happened with me after I deleted that file.

The error message that raspiblitz could look for would match a regex when parsing bitcoin's debug.log, something like this:
`/^Fatal LevelDB error: Corruption: (\d+) missing files; e.g.: \/mnt\/hdd\/bitcoin\/indexes\/txindex\/(\d+).ldb$/`

If such a match was found, the recommended action would be to say that a critical file was missing and that a reindex may be necessary to correct it.
